File: javascript/spawn_agent_f911bf.js
#!/usr/bin/env node
// spawn_agent.py — portiert nach javascript
// Quelle: python, OpenClaw@gateway2:skills/sub-agents-utils/scripts/spawn_agent.py
// Erzeugt: 2026-08-22 durch ABSTRACTIONS_MANAGER.py

/**
 * Sub-Agent spawner - Einfache CLI für sessions_spawn
 */

const fs = require('fs');
const path = require('path');
const { program } = require('commander');

// Simuliere den WORKSPACE-Pfad und füge ihn zum Suchpfad hinzu
const WORKSPACE = '/home/openclaw/.openclaw/workspace';
// In Node.js gibt es keinen direkten sys.path-Ersatz, aber wir können Module laden

// Lade die Modellkonfiguration (simuliert)
let MODELS = [];
try {
    // In einer echten Implementierung würden wir hier die Modelle laden
    // Für dieses Beispiel verwenden wir eine feste Liste
    MODELS = ['openai/gpt-4', 'openai/gpt-3.5-turbo', 'anthropic/claude-3-opus', 'openrouter/anthropic/claude-haiku-4.5'];
} catch (error) {
    console.error(`Modellkonfiguration kann nicht geladen werden: ${error.message}`);
    process.exit(1);
}

class SubAgentSpawner {
    /** Hilft beim Spawnen von Sub-Agents */
    
    static getSpawnConfig(options) {
        /** Erstellt Konfiguration für sessions_spawn */
        
        const config = {
            task: options.task
        };
        
        if (options.label) {
            config.label = options.label;
        }
        if (options.model && MODELS.includes(options.model)) {
            config.model = options.model;
        }
        if (options.thinking) {
            config.thinking = options.thinking;
        }
        if (options.timeout) {
            config.runTimeoutSeconds = options.timeout;
        }
        if (options.thread) {
            config.thread = true;
            if (options.mode === 'run') {
                config.mode = 'session'; // thread requires session mode
            }
        } else {
            config.mode = options.mode;
        }
        
        return config;
    }
    
    static printSpawnCommand(config) {
        /** Gibt das equivalente Tool-Kommando aus */
        console.log('\n🛠️  Tool-Aufruf:');
        console.log('=' .repeat(50));
        console.log('sessions_spawn(');
        for (const [key, value] of Object.entries(config)) {
            if (typeof value === 'string') {
                console.log(`    ${key}="${value}"`);
            } else {
                console.log(`    ${key}=${value}`);
            }
        }
        console.log(')');
        console.log('=' .repeat(50));
    }
    
    static printSlashCommand(config) {
        /** Gibt das equivalente Slash-Kommando aus */
        const task = config.task || '';
        const label = config.label || 'agent';
        const model = config.model || '';
        
        let cmd = `/subagents spawn ${label} "${task}"`;
        if (model) {
            cmd += ` --model ${model}`;
        }
        if (config.thinking) {
            cmd += ` --thinking ${config.thinking}`;
        }
        
        console.log('\n💬 Slash Command:');
        console.log('=' .repeat(50));
        console.log(cmd);
        console.log('=' .repeat(50));
    }
}

function main() {
    program
        .description('Sub-Agent Spawn Helper')
        .option('-t, --task <task>', 'Aufgabenbeschreibung', '')
        .option('-l, --label <label>', 'Optionaler Label')
        .option('-m, --model <model>', 'KI-Modell', MODELS)
        .option('--thinking <level>', 'Thinking Level', ['low', 'medium', 'high'])
        .option('--timeout <seconds>', 'Timeout in Sekunden', 900)
        .option('--thread', 'Thread-Binding aktivieren', false)
        .option('--mode <mode>', 'Run mode', 'run', ['run', 'session'])
        .option('-o, --output <format>', 'Output format', 'tool', ['tool', 'slash', 'json'])
        .addHelpText('afterAll', `
Beispiele:
  spawn_agent.js -t "Analyze logs" 
  spawn_agent.js -t "Code review" -m openrouter/anthropic/claude-haiku-4.5 --timeout 1800
  spawn_agent.js -t "Batch process" -l "batch-worker" --thread
        `);

    program.parse();
    
    const options = program.opts();
    
    if (!options.task) {
        console.error('Fehler: --task ist erforderlich');
        program.help();
        process.exit(1);
    }
    
    const spawner = new SubAgentSpawner();
    const config = spawner.getSpawnConfig(options);
    
    console.log('✅ Sub-Agent Konfiguration:');
    console.log(JSON.stringify(config, null, 2));
    
    if (options.output === 'tool') {
        spawner.printSpawnCommand(config);
    } else if (options.output === 'slash') {
        spawner.printSlashCommand(config);
    } else if (options.output === 'json') {
        console.log('\n📄 JSON:');
        console.log(JSON.stringify(config));
        
        // Speichere als Datei
        const outputDir = '/tmp';
        const fileName = `subagent_${config.label || 'spawn'}.json`;
        const outputPath = path.join(outputDir, fileName);
        
        fs.writeFileSync(outputPath, JSON.stringify(config, null, 2));
        console.log(`💾 Gespeichert: ${outputPath}`);
    }
}

main();

File: perl5/spawn_agent_f911bf.pl
#!/usr/bin/perl
# spawn_agent.py — portiert nach perl5
# Quelle: python, OpenClaw@gateway2:skills/sub-agents-utils/scripts/spawn_agent.py
# Erzeugt: 2026-08-22 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;
use Getopt::Long;
use Pod::Usage;
use JSON;
use File::Spec;
use File::Basename;

=head1 NAME

spawn_agent.pl - Sub-Agent Spawner - Einfache CLI fuer sessions_spawn

=head1 SYNOPSIS

spawn_agent.pl [options]

 Options:
   --task, -t          Aufgabenbeschreibung (required)
   --label, -l         Optionaler Label
   --model, -m         KI-Modell
   --thinking          Thinking Level (low|medium|high)
   --timeout           Timeout in Sekunden (default: 900)
   --thread            Thread-Binding aktivieren
   --mode              Run mode (run|session) (default: run)
   --output, -o        Output format (tool|slash|json) (default: tool)
   --help, -h          Hilfe anzeigen

=cut

# WORKSPACE-Pfad setzen
my $WORKSPACE = "/home/openclaw/.openclaw/workspace";
unshift @INC, $WORKSPACE unless grep { $_ eq $WORKSPACE } @INC;

# Modellkonfiguration laden
my %MODELS;
eval {
    require openclaw_models;
    openclaw_models->import();
    %MODELS = map { $_ => 1 } configured_models();
};
if ($@) {
    die "Modellkonfiguration kann nicht geladen werden: $@\n";
}

# Standardwerte
my $task;
my $label;
my $model;
my $thinking;
my $timeout = 900;
my $thread = 0;
my $mode = "run";
my $output = "tool";
my $help = 0;

GetOptions(
    "task|t=s"      => \$task,
    "label|l=s"     => \$label,
    "model|m=s"     => \$model,
    "thinking=s"    => \$thinking,
    "timeout=i"     => \$timeout,
    "thread"        => \$thread,
    "mode=s"        => \$mode,
    "output|o=s"    => \$output,
    "help|h"        => \$help,
) or pod2usage(2);

pod2usage(1) if $help;

unless ($task) {
    pod2usage("Error: --task ist erforderlich\n");
}

# Modellvalidierung
if ($model && !$MODELS{$model}) {
    die "Ungueltiges Modell: $model. Verfuegbare Modelle: " . join(", ", keys %MODELS) . "\n";
}

# Denken-Level validieren
if ($thinking && $thinking !~ /^(low|medium|high)$/) {
    die "Ungueltiges Thinking Level: $thinking. Moegliche Werte: low, medium, high\n";
}

# Mode validieren
if ($mode !~ /^(run|session)$/) {
    die "Ungueltiger Mode: $mode. Moegliche Werte: run, session\n";
}

# Output-Format validieren
if ($output !~ /^(tool|slash|json)$/) {
    die "Ungueltiges Output-Format: $output. Moegliche Werte: tool, slash, json\n";
}

# SubAgentSpawner-Klasse simulieren
package SubAgentSpawner;

sub new {
    my $class = shift;
    return bless {}, $class;
}

sub get_spawn_config {
    my ($self, %args) = @_;
    
    my %config = (
        task => $args{task}
    );
    
    $config{label} = $args{label} if defined $args{label};
    $config{model} = $args{model} if defined $args{model} && exists $MODELS{$args{model}};
    $config{thinking} = $args{thinking} if defined $args{thinking};
    $config{runTimeoutSeconds} = $args{timeout} if defined $args{timeout};
    
    if ($args{thread}) {
        $config{thread} = JSON::true;
        if ($args{mode} eq "run") {
            $config{mode} = "session";  # thread erfordert session mode
        }
    } else {
        $config{mode} = $args{mode};
    }
    
    return \%config;
}

sub print_spawn_command {
    my ($self, $config) = @_;
    
    print "\n🛠️  Tool-Aufruf:\n";
    print "=" x 50 . "\n";
    print "sessions_spawn(\n";
    
    my @keys = sort keys %$config;
    for my $key (@keys) {
        my $value = $config->{$key};
        if (ref($value) eq 'JSON::XS::Boolean') {
            $value = $value ? 'True' : 'False';
        } elsif (!ref($value)) {
            if ($value =~ /^[0-9]+$/) {
                print "    $key=$value\n";
            } else {
                print "    $key=\"$value\"\n";
            }
        } else {
            print "    $key=$value\n";
        }
    }
    
    print ")\n";
    print "=" x 50 . "\n";
}

sub print_slash_command {
    my ($self, $config) = @_;
    
    my $task = $config->{task} // "";
    my $label = $config->{label} // "agent";
    my $model = $config->{model} // "";
    
    my $cmd = "/subagents spawn $label \"$task\"";
    $cmd .= " --model $model" if $model;
    $cmd .= " --thinking $config->{thinking}" if $config->{thinking};
    
    print "\n💬 Slash Command:\n";
    print "=" x 50 . "\n";
    print "$cmd\n";
    print "=" x 50 . "\n";
}

# Hauptprogramm
package main;

my $spawner = SubAgentSpawner->new();
my $config = $spawner->get_spawn_config(
    task => $task,
    label => $label,
    model => $model,
    thinking => $thinking,
    timeout => $timeout,
    thread => $thread,
    mode => $mode
);

print "✅ Sub-Agent Konfiguration:\n";
print to_json($config, { pretty => 1 }) . "\n";

if ($output eq "tool") {
    $spawner->print_spawn_command($config);
} elsif ($output eq "slash") {
    $spawner->print_slash_command($config);
} elsif ($output eq "json") {
    print "\n📄 JSON:\n";
    print to_json($config) . "\n";
    
    # Speichere als Datei
    my $label_for_filename = $config->{label} // "spawn";
    my $filename = $label_for_filename;
    $filename =~ s/[^\w\-]//g;  # Entferne ungueltige Zeichen
    $filename ||= "spawn";
    
    my $output_file = File::Spec->catfile("/tmp", "subagent_${filename}.json");
    open my $fh, '>', $output_file or die "Kann $output_file nicht schreiben: $!";
    print $fh to_json($config, { pretty => 1 });
    close $fh;
    print "💾 Gespeichert: $output_file\n";
}
